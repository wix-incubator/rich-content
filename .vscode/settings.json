{
  "editor.formatOnSave": true,
  "prettier.disableLanguages": ["vue", "css", "scss", "sass", "markdown"],
  "[json]": {
    "editor.formatOnSave": false
  },
  "eslint.workingDirectories": ["**/src/**"]
}
