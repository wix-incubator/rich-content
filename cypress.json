{
  "chromeWebSecurity": false,
  "baseUrl": "http://localhost:3002",
  "testFiles": "**/*.e2e.js",
  "ignoreTestFiles": [
    "**/__snapshots__/*",
    "**/__image_snapshots__/*"
  ],
  "integrationFolder": "e2e",
  "fixturesFolder": "e2e/cypress/fixtures",
  "pluginsFile": "e2e/cypress/plugins/index.js",
  "screenshotsFolder": "e2e/cypress/screenshots",
  "supportFile": "e2e/cypress/support/index.js",
  "video": true,
  "videoUploadOnPasses": false,
  "videosFolder": "e2e/cypress/videos",
  "env": {},
  "projectId": "6yj6sr"
}
