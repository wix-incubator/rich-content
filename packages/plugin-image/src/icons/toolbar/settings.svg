<?xml version="1.0" encoding="UTF-8" standalone="no"?>
<svg width="19px" height="19px" viewBox="0 0 19 19" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
    <!-- Generator: Sketch 47.1 (45422) - http://www.bohemiancoding.com/sketch -->
    <title>01. Icons / General / Settings / 01. Default</title>
    <desc>Created with Sketch.</desc>
    <defs>
        <path d="M16.9553,8.6584 C17.0183,9.2294 17.0143,9.8044 16.9463,10.3744 C16.9203,10.5934 16.7283,10.7544 16.5073,10.7414 L16.4133,10.7414 C15.7273,10.7394 15.1103,11.1604 14.8623,11.8014 C14.6153,12.4414 14.7893,13.1684 15.2993,13.6274 C15.4633,13.7754 15.4833,14.0254 15.3443,14.1974 C14.9873,14.6424 14.5803,15.0444 14.1313,15.3954 C13.9603,15.5274 13.7163,15.5074 13.5693,15.3494 C13.0903,14.8494 12.3603,14.6834 11.7123,14.9264 C11.0713,15.1934 10.6623,15.8304 10.6873,16.5254 C10.6953,16.7444 10.5313,16.9324 10.3123,16.9544 C10.0393,16.9844 9.7643,17.0004 9.4893,17.0004 C9.2003,16.9994 8.9123,16.9834 8.6253,16.9504 C8.4063,16.9244 8.2443,16.7324 8.2573,16.5114 C8.2973,15.8094 7.8913,15.1584 7.2433,14.8854 C6.5933,14.6344 5.8553,14.8004 5.3743,15.3054 C5.2263,15.4694 4.9763,15.4894 4.8043,15.3514 C4.3643,14.9974 3.9663,14.5944 3.6173,14.1504 C3.4843,13.9794 3.5043,13.7354 3.6633,13.5884 C4.1763,13.1184 4.3433,12.3794 4.0823,11.7344 C3.8303,11.1054 3.2173,10.6964 2.5393,10.7034 C2.3103,10.7094 2.1073,10.5564 2.0483,10.3354 C1.9843,9.7684 1.9843,9.1964 2.0483,8.6294 C2.0693,8.4104 2.2613,8.2474 2.4813,8.2624 C3.1803,8.2844 3.8233,7.8844 4.1133,7.2484 C4.3853,6.5984 4.2173,5.8474 3.6943,5.3744 C3.5303,5.2264 3.5103,4.9764 3.6493,4.8054 C4.0063,4.3604 4.4133,3.9574 4.8623,3.6054 C5.0333,3.4744 5.2773,3.4944 5.4243,3.6534 C5.9033,4.1524 6.6333,4.3184 7.2813,4.0764 C7.9253,3.8104 8.3363,3.1724 8.3113,2.4764 C8.3043,2.2574 8.4683,2.0704 8.6863,2.0474 C9.2473,1.9844 9.8133,1.9844 10.3733,2.0474 C10.5933,2.0724 10.7543,2.2654 10.7423,2.4864 C10.7013,3.1874 11.1073,3.8384 11.7543,4.1124 C12.4063,4.3654 13.1463,4.1994 13.6283,3.6924 C13.7763,3.5294 14.0263,3.5094 14.1983,3.6474 C14.6383,4.0004 15.0373,4.4024 15.3863,4.8464 C15.5203,5.0174 15.5003,5.2614 15.3403,5.4084 C14.8313,5.8704 14.6613,6.5994 14.9123,7.2404 C15.1643,7.8804 15.7863,8.2984 16.4733,8.2894 C16.7003,8.2854 16.9003,8.4384 16.9553,8.6584 Z M15.9933,9.7754 C16.0013,9.6004 16.0023,9.4254 15.9963,9.2514 C15.0913,9.0954 14.3273,8.4844 13.9823,7.6064 C13.6513,6.7654 13.7813,5.8154 14.2883,5.0964 C14.1753,4.9724 14.0563,4.8534 13.9343,4.7404 C13.4803,5.0544 12.9353,5.2294 12.3773,5.2294 C12.0393,5.2294 11.7083,5.1674 11.3653,5.0334 C10.5113,4.6724 9.9183,3.8964 9.7733,3.0044 C9.6923,3.0024 9.6113,3.0004 9.5303,3.0004 C9.4433,3.0004 9.3573,3.0024 9.2693,3.0054 C9.1093,3.8864 8.5053,4.6524 7.6313,5.0124 C7.3243,5.1284 7.0023,5.1854 6.6743,5.1854 C6.1193,5.1854 5.5763,5.0134 5.1183,4.6974 C4.9903,4.8144 4.8663,4.9364 4.7473,5.0634 C5.2723,5.8024 5.3923,6.7804 5.0233,7.6644 C4.6523,8.4754 3.9103,9.0474 3.0063,9.2134 C2.9983,9.3884 2.9983,9.5644 3.0053,9.7404 C3.9023,9.8924 4.6633,10.4954 5.0093,11.3594 C5.3513,12.2074 5.2273,13.1694 4.7143,13.8984 C4.8273,14.0224 4.9453,14.1424 5.0683,14.2574 C5.5213,13.9434 6.0663,13.7694 6.6243,13.7694 C6.9593,13.7694 7.2883,13.8314 7.6323,13.9624 C8.4873,14.3244 9.0813,15.1004 9.2253,15.9954 C9.3133,15.9984 9.4013,16.0004 9.4893,16.0004 C9.5693,16.0004 9.6483,15.9984 9.7283,15.9954 C9.8893,15.1174 10.4903,14.3514 11.3613,13.9904 C11.6693,13.8754 11.9923,13.8164 12.3203,13.8164 C12.8753,13.8164 13.4183,13.9884 13.8753,14.3044 C14.0043,14.1874 14.1273,14.0654 14.2463,13.9384 C13.7383,13.2224 13.6063,12.2784 13.9303,11.4394 C14.2783,10.5424 15.0593,9.9214 15.9933,9.7754 Z M9.5,12 C8.11928813,12 7,10.8807119 7,9.5 C7,8.11928813 8.11928813,7 9.5,7 C10.8807119,7 12,8.11928813 12,9.5 C12,10.8807119 10.8807119,12 9.5,12 Z M9.5,11 C10.3284271,11 11,10.3284271 11,9.5 C11,8.67157288 10.3284271,8 9.5,8 C8.67157288,8 8,8.67157288 8,9.5 C8,10.3284271 8.67157288,11 9.5,11 Z" id="settings-path"></path>
    </defs>
    <g id="Symbols" stroke="none" stroke-width="1" fill-rule="evenodd">
        <g id="01.-Icons-/-General-/-Settings-/-01.-Default">
            <mask id="settings-mask">
                <use xlink:href="#path-1"></use>
            </mask>
            <use id="settings-use" fill-rule="nonzero" xlink:href="#path-1"></use>
        </g>
    </g>
</svg>
