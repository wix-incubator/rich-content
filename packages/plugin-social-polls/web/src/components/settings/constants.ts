export const TABS = {
  LAYOUT: 'layout',
  DESIGN: 'design',
  SETTINGS: 'settings',
  EDIT: 'edit',
};
