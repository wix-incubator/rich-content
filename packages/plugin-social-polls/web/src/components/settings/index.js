export * from './SettingsModal';
export * from './constants';
