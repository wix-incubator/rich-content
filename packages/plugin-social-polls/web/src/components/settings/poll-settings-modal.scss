@import "~wix-rich-content-common/dist/statics/styles/mixins";

.header {
  padding: 0 20px;
  border-bottom: 1px solid #ededed;

  .title {
    margin: 25px 0 20px;
    font-size: 20px;
    line-height: 1.4;
  }

  @include phone {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    align-items: center;
    padding: 20px;
  }
}

.header_button_list {
  display: flex;
  height: auto;
  overflow: hidden;
  align-items: center;
}

.poll_header_button {
  border: 0;
  outline: none;
  background: none;
  font-size: 16px;
  line-height: 1;
  padding: 0;
  font-family: inherit;
  display: flex;
  flex-direction: row;
  align-items: center;

  &.poll_header_button_primary {
    color: #0261ff;
  }
  
  &.poll_header_button_secondary {
    color: inherit;
  }
}

.section {
  margin: 0 -10px;
}

.title {
  font-size: 16px;
  line-height: 1.5;
  margin: 0 0 15px;
}

.subtitle {
  font-size: 14px;
  line-height: 1.43;
  opacity: 0.6;
  margin: 0 0 5px;
}

.sliderWithInput_label,
.label {
  font-size: 14px;
  line-height: 1.43;
  margin: 0 0 5px;
}

.sliderWithInput_label {
  margin: 10px 0;
}

.labeled_toggle_label,
.checkbox_label {
  @extend .label;
}

.preset_selector {
  grid-template-columns: repeat(3, 1fr);
  grid-column-gap: 10px;
  margin: 10px 0;
}

.layout_selector {
  grid-template-columns: repeat(4, 1fr);
  grid-auto-rows: 1fr;
  grid-column-gap: 10px;
  margin: 20px 0;

  @include phone {
    grid-template-columns: repeat(3, 1fr);
  }
}

.radioGroup {
  display: flex;
  flex-direction: row;
  align-items: center;
  margin: 5px 0;
  
  &.radioPanel {
    margin: 10px 0 20px;
    display: flex;
    flex-direction: column;
    align-items: baseline;
  }
}

.selectionListOption {
  margin: 0;
  color: #000;
  padding-top: 10px;
  display: flex;
  flex-direction: column;
  align-items: center;
  height: auto;
  box-sizing: border-box;
}

.selectionListOptionLabel {
  margin: 5px 0;
}

.selectionListOption:hover:not(.selectionListOption_selected) {
  background-color: rgba(2, 97, 255, 0.05);
}

.selectionListOption_selected {
  background-color: rgba(2, 97, 255, 0.2);
}

.separator {
  margin: 20px 0;
}

.separator_vertical {
  height: 26px;
  margin: 0 15px;
}

.labeled_toggle_root {
  padding: 0;
  margin: 15px 0;
  cursor: pointer;

  @include phone {
    padding-top: 0 !important;
  }
}

.modal {
  background: transparent;
  color: #000;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  width: 80%;
}

.overlay {
  position: fixed;
  background-color: rgba(0, 0, 0, 0.6);
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  z-index: 8;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  cursor: auto;
}