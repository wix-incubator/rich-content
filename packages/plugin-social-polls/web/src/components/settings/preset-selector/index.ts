export * from './PollPresetSelector';
export * from './Arrow';
