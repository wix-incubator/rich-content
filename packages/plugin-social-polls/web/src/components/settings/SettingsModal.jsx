import React, { Component } from 'react';
import PropTypes from 'prop-types';
import cls from 'classnames';
import ReactModal from 'react-modal';

import {
  SettingsPanelFooter,
  Tabs,
  Tab,
  FocusManager,
  Separator,
} from 'wix-rich-content-editor-common';
import { mergeStyles } from 'wix-rich-content-common';

import { PollContextProvider } from '../poll-context';
import { RCEHelpersContext } from '../rce-helpers-context';

import { TABS } from './constants';

import { DesignSettingsSection } from './design-settings-section';
import { LayoutSettingsSection } from './layout-settings-section';
import { PollSettingsSection } from './poll-settings-section';
import { EditPollSection } from './edit-poll-section';
import { PollViewer } from '../PollViewer';

import styles from './poll-settings-modal.scss';

export class SettingsModal extends Component {
  static propTypes = {
    activeTab: PropTypes.string.isRequired,
    componentData: PropTypes.object.isRequired,
    helpers: PropTypes.object.isRequired,
    pubsub: PropTypes.any.isRequired,
    isMobile: PropTypes.bool,
    theme: PropTypes.object.isRequired,
    t: PropTypes.func,
    relValue: PropTypes.string,
    anchorTarget: PropTypes.string,
    languageDir: PropTypes.string,
  };

  static defaultProps = {
    activeTab: TABS.LAYOUT,
  };

  state = {
    activeTab: this.props.activeTab,
    componentData: this.props.componentData,
    $container: React.createRef(),
    isPreviewOpen: false,
  };

  styles = mergeStyles({ styles, theme: this.props.theme });

  setPoll = poll => {
    const { pubsub } = this.props;

    const componentData = pubsub.store.get('componentData');

    pubsub.store.set('componentData', {
      ...componentData,
      poll,
    });
  };

  componentDidCatch() {}

  handleTabChange = activeTab => this.setState({ activeTab });

  getContainer = () => this.state.$container.current;

  closePreview = () => this.setState({ isPreviewOpen: false });

  openPreview = () => this.setState({ isPreviewOpen: true });

  restoreChanges = () => {
    const { pubsub, helpers } = this.props;
    const { componentData } = this.state;

    pubsub.set('componentData', componentData);

    helpers.closeModal();
  };

  componentDidMount() {
    this.props.pubsub.subscribe('componentData', this.onComponentUpdate);
  }

  componentWillUnmount() {
    this.props.pubsub.unsubscribe('componentData', this.onComponentUpdate);
  }

  onComponentUpdate = () => this.forceUpdate();

  render() {
    const { activeTab, $container, isPreviewOpen } = this.state;
    const { pubsub, helpers, t, languageDir, theme, isMobile } = this.props;

    const componentData = pubsub.store.get('componentData');

    return (
      <div ref={$container}>
        <FocusManager dir={languageDir}>
          {isMobile ? (
            <div className={this.styles.header}>
              <button
                className={cls(
                  this.styles.poll_header_button,
                  this.styles.poll_header_button_secondary
                )}
                onClick={this.restoreChanges}
              >
                {t('Poll_PollSettings_Common_CTA_Secondary')}
              </button>
              <div className={this.styles.header_button_list}>
                <button
                  className={cls(
                    this.styles.poll_header_button,
                    this.styles.poll_header_button_primary
                  )}
                  onClick={this.openPreview}
                >
                  {t('Poll_FormatToolbar_Preview_Tooltip')}
                </button>
                <Separator className={this.styles.separator_vertical} />
                <button
                  className={cls(
                    this.styles.poll_header_button,
                    this.styles.poll_header_button_primary
                  )}
                  onClick={helpers.closeModal}
                >
                  {t('Poll_PollSettings_Common_CTA_Primary')}
                </button>
              </div>
            </div>
          ) : (
            <div className={this.styles.header}>
              <h3 className={this.styles.title}>{t('Poll_PollSettings_Common_Header')}</h3>
            </div>
          )}

          {activeTab === TABS.EDIT ? (
            <RCEHelpersContext.Provider
              value={{
                layout: componentData.layout,
                design: componentData.design,
                helpers,
                theme,
                t,
                isMobile,
              }}
            >
              <PollContextProvider
                settings={{}}
                poll={componentData.poll}
                setPoll={this.setPoll}
                t={t}
              >
                <EditPollSection store={pubsub.store} />
              </PollContextProvider>
            </RCEHelpersContext.Provider>
          ) : (
            <Tabs value={activeTab} theme={theme} onTabSelected={this.handleTabChange}>
              <Tab
                label={t('Poll_PollSettings_Tab_Layout_TabName')}
                value={TABS.LAYOUT}
                theme={theme}
              >
                <LayoutSettingsSection
                  theme={theme}
                  store={pubsub.store}
                  componentData={componentData}
                  t={t}
                  isMobile={isMobile}
                />
              </Tab>
              <Tab
                label={t('Poll_PollSettings_Tab_Design_TabName')}
                value={TABS.DESIGN}
                theme={theme}
              >
                <DesignSettingsSection
                  theme={theme}
                  store={pubsub.store}
                  componentData={componentData}
                  t={t}
                />
              </Tab>
              <Tab
                label={t('Poll_PollSettings_Tab_Settings_TabName')}
                value={TABS.SETTINGS}
                theme={theme}
              >
                <PollSettingsSection
                  theme={theme}
                  store={pubsub.store}
                  componentData={componentData}
                  t={t}
                />
              </Tab>
            </Tabs>
          )}

          {!isMobile && (
            <SettingsPanelFooter
              fixed
              cancel={this.restoreChanges}
              save={helpers.closeModal}
              theme={this.props.theme}
              t={t}
            />
          )}

          {isMobile && $container.current && (
            <ReactModal
              isOpen={isPreviewOpen}
              onRequestClose={this.closePreview}
              parentSelector={this.getContainer}
              className={cls(this.styles.modal)}
              overlayClassName={cls(this.styles.overlay)}
            >
              <PollViewer
                isMobile
                settings={{
                  isPreview: true,
                  preventInteraction: true,
                }}
                componentData={componentData}
                onRequestClose={this.closePreview}
                theme={theme}
                t={t}
              />
              &nbsp;
              <button
                className={cls(
                  this.styles.poll_header_button,
                  this.styles.poll_header_button_primary
                )}
                onClick={this.closePreview}
              >
                {t('Poll_Preview_Close_CTA')}
              </button>
            </ReactModal>
          )}
        </FocusManager>
      </div>
    );
  }
}
