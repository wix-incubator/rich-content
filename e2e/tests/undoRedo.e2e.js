/* eslint-disable max-len */
/*global cy*/
import { IMAGE_SETTINGS, PLUGIN_COMPONENT } from '../cypress/dataHooks';
import { DEFAULT_DESKTOP_BROWSERS } from './settings';
import { usePlugins, plugins } from '../cypress/testAppConfig';

const eyesOpen = ({
  test: {
    parent: { title },
  },
}) =>
  cy.eyesOpen({
    appName: 'Plugins',
    testName: title,
    browser: DEFAULT_DESKTOP_BROWSERS,
  });

describe('plugins', () => {
  //   afterEach(() => cy.matchContentSnapshot());

  context('undo redo', () => {
    before(function() {
      eyesOpen(this);
    });

    beforeEach('load editor', () => {
      cy.switchToDesktop();
    });

    after(() => cy.eyesClose());

    it('should undo and redo image plugin customizations', function() {
      cy.loadRicosEditorAndViewer('empty', usePlugins(plugins.all));
      cy.addImage();
      cy.enterParagraphs([' testing undo redo for plugins']);
      cy.openImageSettings();
      cy.get(`[data-hook=${IMAGE_SETTINGS.PREVIEW}]:first`);
      cy.addImageTitle();
      cy.undo();
      cy.get('div').should('not.have.text', 'Title');
      cy.undo();
      cy.get('span').should('not.have.text', 'testing undo redo for plugins');
      cy.undo();
      cy.get(`[data-hook=${PLUGIN_COMPONENT.IMAGE}]:first`).should('not.exist');
      cy.redo();
      cy.get(`[data-hook=${PLUGIN_COMPONENT.IMAGE}]:first`).should('exist');
      cy.redo();
      cy.get('.public-DraftStyleDefault-block > [data-offset-key="2s2ri-0-0"] > span').should(
        'have.text',
        'testing undo redo for plugins'
      );
      cy.redo();
      cy.get('input').should('have.value', 'Title');
      cy.eyesCheckWindow(this.test.title);
    });

    it('should undo and redo accordion plugin customizations', function() {
      cy.loadRicosEditorAndViewer('empty', usePlugins(plugins.all));
      cy.addAccordion();
      cy.focusAccordion(1).type('Yes\n');
      cy.addAccordionPair();
      cy.focusAccordion(2).insertPluginFromSideToolbar('ImagePlugin_InsertButton');
      cy.undo()
        .undo()
        .undo()
        .undo()
        .undo()
        .undo()
        .undo();
      cy.redo()
        .redo()
        .redo()
        .redo();
      cy.redo()
        .redo()
        .redo();
      cy.eyesCheckWindow(this.test.title);
    });
  });
});
