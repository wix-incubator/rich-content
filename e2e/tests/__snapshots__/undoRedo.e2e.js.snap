exports[`plugins > undo redo > should undo and redo image plugin customizations #0`] =
{
  "blocks": [
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 0,
      "text": "testing undo redo for plugins",
      "type": "unstyled"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [
        {
          "key": 0,
          "length": 1,
          "offset": 0
        }
      ],
      "inlineStyleRanges": [],
      "key": 1,
      "text": " ",
      "type": "atomic"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 2,
      "text": "",
      "type": "unstyled"
    }
  ],
  "entityMap": {
    "0": {
      "data": {
        "config": {
          "alignment": "center",
          "disableExpand": false,
          "showDescription": true,
          "showTitle": true,
          "size": "content"
        },
        "metadata": {
          "caption": "Title"
        },
        "src": {
          "file_name": "8bb438_1b73a6b067b24175bd087e86613bd00c.jpg",
          "height": 1000,
          "id": "8b72558253b2502b401bb46e5599f22a",
          "original_file_name": "8bb438_1b73a6b067b24175bd087e86613bd00c.jpg",
          "width": 1920
        }
      },
      "mutability": "IMMUTABLE",
      "type": "wix-draft-plugin-image"
    }
  }
};

exports[`plugins > undo redo > should undo and redo accordion plugin customizations #0`] =
{
  "blocks": [
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 0,
      "text": "",
      "type": "unstyled"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [
        {
          "key": 0,
          "length": 1,
          "offset": 0
        }
      ],
      "inlineStyleRanges": [],
      "key": 1,
      "text": " ",
      "type": "atomic"
    },
    {
      "data": {},
      "depth": 0,
      "entityRanges": [],
      "inlineStyleRanges": [],
      "key": 2,
      "text": "",
      "type": "unstyled"
    }
  ],
  "entityMap": {
    "0": {
      "data": {
        "config": {
          "direction": "ltr",
          "expandOnlyOne": false,
          "expandState": "first_expanded"
        },
        "pairs": [
          {
            "content": {
              "blocks": [
                {
                  "data": {},
                  "depth": 0,
                  "entityRanges": [],
                  "inlineStyleRanges": [],
                  "key": 0,
                  "text": "",
                  "type": "unstyled"
                },
                {
                  "data": {},
                  "depth": 0,
                  "entityRanges": [
                    {
                      "key": 0,
                      "length": 1,
                      "offset": 0
                    }
                  ],
                  "inlineStyleRanges": [],
                  "key": 1,
                  "text": " ",
                  "type": "atomic"
                },
                {
                  "data": {},
                  "depth": 0,
                  "entityRanges": [],
                  "inlineStyleRanges": [],
                  "key": 2,
                  "text": "",
                  "type": "unstyled"
                }
              ],
              "entityMap": {
                "0": {
                  "data": {
                    "config": {
                      "alignment": "center",
                      "disableExpand": false,
                      "showDescription": true,
                      "showTitle": true,
                      "size": "content"
                    },
                    "src": {
                      "file_name": "8bb438_1b73a6b067b24175bd087e86613bd00c.jpg",
                      "height": 1000,
                      "id": "8b72558253b2502b401bb46e5599f22a",
                      "original_file_name": "8bb438_1b73a6b067b24175bd087e86613bd00c.jpg",
                      "width": 1920
                    }
                  },
                  "mutability": "IMMUTABLE",
                  "type": "wix-draft-plugin-image"
                }
              }
            },
            "key": "syzz39rcc",
            "title": {
              "blocks": [
                {
                  "data": {},
                  "depth": 0,
                  "entityRanges": [],
                  "inlineStyleRanges": [],
                  "key": 0,
                  "text": "Yes ",
                  "type": "unstyled"
                }
              ],
              "entityMap": {}
            }
          },
          {
            "content": {
              "blocks": [
                {
                  "data": {},
                  "depth": 0,
                  "entityRanges": [],
                  "inlineStyleRanges": [],
                  "key": 0,
                  "text": "",
                  "type": "unstyled"
                }
              ],
              "entityMap": {}
            },
            "key": "pbd8wx06t",
            "title": {
              "blocks": [
                {
                  "data": {},
                  "depth": 0,
                  "entityRanges": [],
                  "inlineStyleRanges": [],
                  "key": 0,
                  "text": "",
                  "type": "unstyled"
                }
              ],
              "entityMap": {}
            }
          }
        ]
      },
      "mutability": "IMMUTABLE",
      "type": "wix-rich-content-plugin-accordion"
    }
  }
};
